src/lib/WalletInfoCollector.ts

```diff
@@ -149,7 +149,10 @@ export default class WalletInfoCollector {
 
     private static async _getWalletInfoInstance(walletId: string, walletType: WalletType): Promise<WalletInfo> {
         const existingWalletInfo = await WalletStore.Instance.get(walletId);
-        if (existingWalletInfo) return existingWalletInfo;
+        if (existingWalletInfo) {
+            existingWalletInfo.keyMissing = false;
```

Comment:
Why do we know this here?
Does this happen after login?

Reply:
Yes, exactly. Just in case you are importing a previously deleted key.

However, this brings me to another problem that I think I found: because keys in the KG are incremental, the ID that gets returned from import is another one than the one requested and not found. So I don't think that the collector even finds the previous meta data here...

This needs comparing of requested and new IDs and an update of the store (and thus also an update of the store in the apps, like Safe). All because of the incremental key IDs...

Reply:
Or rather, the collector _does_ find the correct wallet info, but fails to update the ID currently, so on the next request the keyguard ID is again not found.

Reply:
And this change if ID bubbles up to the apps using the AM, as now they also need to know about and handle the if change.

Reply:
Not nice indeed...
However, instead of changing existing records with the new key id, the accounts manager could also map the not anymore existing id to the new id in all requests.
On reimports, the labels of the old account info can be copied to the new one.
This way, at least apps like the safe wouldn't need to handle this and can continue submitting requests with the old id.

Both approaches result in sort of a mess though.

Reply:
It would be possible to add the old keyId to the login request. In keyguard we would check if the id is still free, and if yes, save the reimported key with the given id.

Reply:
This is resolved with #205.